docs/user-guide/tips-tricks-and-anti-patterns.ipynb

```diff
@@ -0,0 +1,278 @@
+    "\n",
+    "In principle the order of dimension in scipp can be arbitrary since operations transpose automatically based in dimension labels.\n",
+    "In practice however a wrong choice of dimension order can lead to performance bottlenecks.\n",
+    "This is most obvious when slicing multi-dimensional variables or arrays, where slicing any but the outer dimension yields a slice with gaps between data values, i.e., a very inefficient memory layout.\n",
+    "If an application requires slicing (directly or indirectly, e.g., in `groupby` operations) predominantly for a certain dimension, this dimension should be made the *outermost* dimension.\n",
+    "For example, for a stack of images the best choice would typically be"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nimage = 13\n",
+    "images = sc.Variable(dims=['image', 'y', 'x'], shape=[nimage, ny, nx,])\n",
+    "images"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Slices such as"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images['image', 3]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "will then have data for all pixels in a contiguous chunk of memory.\n",
+    "Note that in scipp the first listed dimension in `dims` is always the *outermost* dimension, i.e. numpy's default.\n",
+    "\n",
+    "## Avoid loops\n",
+    "\n",
+    "With scipp, just like with numpy or Matlab, loops such as `for`-loops should be avoided.\n",
+    "Loops typically lead to many small slices or many small array objects and very quickly lead to very inefficient code.\n",
+    "If we encounter the need for a loop in a workflow using scipp we should try and take a step back to understand how it can be avoided.\n",
+    "Some tips to do this include:\n",
+    "\n",
+    "### Use slicing with \"shifts\"\n",
+    "\n",
+    "When access to neighbor slices is required, replace"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for i in range(len(images.values)-1):\n",
+    "    images['image', i] -= images['image', i+1]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "with"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images['image', :-1] -= images['image', 1:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that a this point numpy provides more powerful functions such as [numpy.roll](https://numpy.org/doc/stable/reference/generated/numpy.roll.html).\n",
+    "Scipp's toolset for such purposes is not fully developed yet."
+   ]
+  },
+  {
+   "cell_type": "markdown",
```

Comment:
It seems obvious but I think the caveat that you are sacrificing unit operations, uncertainty handling... should be reminded.